detd/sysctl.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(C) 2024 Intel Corporation
+# Authors:
+#   Hector Blanco Alcaine
+
+""" Module sysctl
+
+This module provides a class to execute sysctl commands.
+
+"""
+
+
+import subprocess
+
+from .common import CommandString
+
+
+
+
+class CommandSysctl:
+
+    def __init__(self):
+        pass
+
+    def run(self, command):
+        cmd = command.split()
+        result = subprocess.run(cmd, capture_output=True, text=True)
+
+        success_codes = [0]
+        if result.returncode not in success_codes:
+            raise subprocess.CalledProcessError(result.returncode, command, result.stdout, result.stderr)
+
+        return result
+
+    def disable_ipv6vlan(self, interface, vid):
+
+        interface_name = f"{interface.name}/{vid}"
+
+        cmd = CommandStringSysctlDisableIpv6(interface_name)
+
+        self.run(cmd)
+
+    def disable_ipv6(self, interface):
```

Comment:
Please use a single function "disable_ipv6" that takes two parameters: interface, vid. When no vlan id is provided, the function will disable ipv6 fo the parent interface, and when provided for the vlan interface.